compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -462,6 +462,11 @@ public void visit(BLangFunction funcNode, AnalyzerData data) {
         validateNamedWorkerUniqueReferences(data);
     }
 
+    private boolean isMethodInServiceDeclaration(BLangFunction func) {
+        BLangNode parent = func.parent;
+        return parent.getKind() == NodeKind.CLASS_DEFN && ((BLangClassDefinition) parent).isServiceDecl;
```

Comment:
On Update 1, the error/warning wasn't logged for service classes either, right? Wasn't part of the discussion, so can you please share why we are only checking for service declarations?

Reply:
Please check https://github.com/ballerina-platform/ballerina-lang/issues/38460#issuecomment-1296572920

Reply:
Yeah, saw that, but my concern is **why** it is only for service declarations? Just wanted to know the reason, if it was discussed.

@hasithaa, would appreciate your input.

Reply:
There was a change to make resource methods public and because of that, this warning appeared for service-decl. But from the client side, we only need to consider the shape of the parameters of resource methods, not the actual type. Therefore it is decided to disallow this warning only for service-decl. We need this warning to client-decl.

Reply:
By client-decl do you mean client classes?

My question is about why we won't be removing the warning for service classes, not client classes.

E.g.,
```ballerina
service class Service {
    isolated resource function get processRequest(Foo foo) returns json { // still a warning
        return {output: "Hello"};
    }
}
```

I'm OK with having the warning here until the rules are finalized since unlike with service declarations, these can be referred to.

Was just asking because this is also a scenario where there was no error/warning previously, but there is one now.

Reply:
@MaryamZi last example, shouldn't be a warning, because the service class is not public. We need to fix this too.

Reply:
Disallowed this warning for service class.